tests/test_id_insertion.py

```diff
@@ -0,0 +1,109 @@
+import assertpy
+import pytest
+from assertpy import assert_that
+
+from metadata_schema import MetadataSchema, MetadataSchemaError
+
+
+def test__inserts_id_property__when_version_from_path_is_found(test_dir, version_map, schema_base_url):
+    # given
+    schema_json = {
+        '$schema': 'http://json-schema.org/draft-07/schema#',
+    }
+    version_map = {
+        'version_numbers': {
+            'type': {
+                'biomaterial': {
+                    'donor': '1.2.3'
+                }
+            }
+        }
+    }
+    schema_path = 'type/biomaterial/donor'
+    # when
+    schema = MetadataSchema(schema_json, schema_path).insert_id_url(version_map, schema_base_url)
+    # then
+    assert schema.get('$id') == 'https://schema.humancellatlas.org/type/biomaterial/1.2.3/donor'
+
+
+def test__raises_error__when_version_from_path_is_not_found(test_dir, version_map, schema_base_url):
+    # given
+    schema_json = {
+        '$schema': 'http://json-schema.org/draft-07/schema#',
+    }
+    version_map = {
+        'version_numbers': {
+            'type': {
+                'biomaterial': {
+                    'donor': '1.2.3'
+                }
+            }
+        }
+    }
+    schema_path = 'type/biomaterial/cell_suspension'
+
+    # when/then
+    with pytest.raises(MetadataSchemaError):
+        MetadataSchema(schema_json, schema_path).get_json_schema(version_map, schema_base_url)
+
+
+def test__inserts_old_id_key__when_draft_04(test_dir, version_map, schema_base_url):
+    # given
+    schema_json = {
+        '$schema': 'http://json-schema.org/draft-04/schema#',
+    }
+    version_map = {
+        'version_numbers': {
+            'type': {
+                'biomaterial': {
+                    'donor': '1.2.3'
+                }
+            }
+        }
+    }
+    schema_path = 'type/biomaterial/donor'
+    # when
+    schema = MetadataSchema(schema_json, schema_path).insert_id_url(version_map, schema_base_url)
+    # then
+    assert schema.get('id') == 'https://schema.humancellatlas.org/type/biomaterial/1.2.3/donor'
+
+from file_util import load_json
```

Comment:
shouldn't all imports be on top? is there a reason why it's here?

Reply:
mistake using auto import